arangod/Aql/OptimizerRulesCluster.cpp

```diff
@@ -230,12 +231,6 @@ bool substituteClusterSingleDocumentOperationsIndex(Optimizer* opt,
           continue;
         }
 
-        if (mod->getOptions().exclusive) {
```

Comment:
we would allow for exclusivity because the transcation for this optimization rule should be el cheapo, so it would go all or nothing.
The exclusive lock has a timeout, so it would stop the deadlock from happening because one would either fail or commit and release the lock and the next one would acquire it

Reply:
@cpjulia I am not sure if this statement is correct. We haven't changed anything in the `SingleRemoteModificationExecutor`, in particular we do not create an el cheapo transaction (though that might not actually be necessary for a single operation). Either way, this change should probably be reverted.

Reply:
Oh ok I was mistakenly thinking about the multiple case. In this case, this if should remain here, I put it back.